docs/references/api.md

```diff
@@ -0,0 +1,277 @@
+# Pro API Reference Guide
+
+The Pro Client has a Python-based API to be consumed by users who want to integrate the client's functionality to their software.
+The functions and objects are available through the `uaclient.api` module, and all of the available endpoints return an object with specific data for the calls.
+
+Besides importing the Python code directly, consumers who are not writing Python may use the CLI to call the same functionality, using the `pro api` command. This command will always return a JSON with a standard structure, as can be seen below:
+
+```json
+{
+    "_schema_version": "v1",
+    "data": {
+        "attributes": {
+            <endpoint specific attributes>
+        },
+        "meta": {
+            "environment_vars": []
+        },
+        "type": "<response_type>"
+    },
+    "errors": [], 
+    "result": "<success|failure>",
+    "version": "<client version>", 
+    "warnings": []
+}
+```
+
+The currently available endpoints are:
+
+## u.pro.version.v1
+Shows the installed Client version.
+
+### Args
+This endpoint takes no arguments.
+
+### Python API interaction
+#### Calling from Python code
+```python
+from uaclient.api.u.pro.version.v1 import version
+
+result = version()
+```
+
+#### Expected return object:
+`uaclient.api.u.pro.version.v1.VersionResult`
+
+|Field Name|Type|Description|Required|
+|-|-|-|-|
+|installed_version|str|The current installed version|Yes|
+
+### Raised Exceptions
+- `VersionError`: raised if the client cannot determine the version
+
+
+### CLI interaction
+#### Calling from the CLI:
+```bash
+pro api u.pro.version.v1
+```
+
+#### Expected JSON output:
+```json
+"attributes": {
+    "installed_version":"<version>"
+}
+```
```

Comment:
Maybe just tweak this header to make it clear these attributes are part of the bigger structure at the top
```suggestion
#### Expected attributes in JSON structure
```json
{
    "installed_version": "<version>"
}
```
```

Reply:
makes sense